src/dev/build/tasks/bundle_fleet_packages.ts

```diff
@@ -61,6 +78,39 @@ export const BundleFleetPackages: Task = {
             skipChecksumCheck: true,
             maxAttempts: 3,
           });
+
+          // If we're force aligning the version, we need to
+          // 1. Unzip the downloaded archive
+          // 2. Edit the `manifest.yml` file to include the updated `version` value
+          // 3. Re-zip the archive and replace it on disk
+          if (fleetPackage.forceAlignStackVersion) {
+            const buffer = await fs.readFile(destination);
+            const zipEntries = await unzipBuffer(buffer);
+
+            const manifestPath = `${fleetPackage.name}-${fleetPackage.version}/manifest.yml`;
+            const manifestEntry = zipEntries.find((entry) => entry.path === manifestPath);
+
+            if (!manifestEntry || !manifestEntry.buffer) {
+              log.debug(
+                `Unable to find manifest.yml for stack aligned package ${fleetPackage.name}`
+              );
+
+              return;
+            }
+
+            const manifestYml = await safeLoad(manifestEntry.buffer.toString('utf8'));
+            manifestYml.version = stackVersion;
+
+            const newManifestYml = safeDump(manifestYml);
+            manifestEntry.buffer = Buffer.from(newManifestYml, 'utf8');
+
+            // Update all paths to use the new version
+            zipEntries.forEach(
+              (entry) => (entry.path = entry.path.replace(fleetPackage.version, versionToWrite!))
+            );
+
+            await createZipFile(zipEntries, destination);
+          }
         } catch (error) {
           log.warning(`Failed to download bundled package archive ${archivePath}`);
```

Comment:
I'm a little concerned about the large try block with only an error log.  Any download, zip, unzip, and version update errors will still produce a successful build.  I've seen several 404s on packages, are we sure this is okay?

Reply:
Fair point. I think we should probably throw and fail the build on a 404 for a bundled package. It should be caught in a PR when the build task runs.

Reply:
404s are indeed concerning. Could you please share more details about where/what have you noticed?

Reply:
I've update this to re-throw the error and prevent a successful build if a bundled package fails to download, unzip, rewrite version, etc

```
❯ yarn build                                                                                                                                               
yarn run v1.22.19
$ node scripts/build --all-platforms
 info [  kibana  ] Bundling fleet packages
   │ info Fetching fleet packages from package registry
       │ debg Found configured bundled packages: apm-8.4.0-fake-version, elastic_agent-1.3.5, endpoint-8.4.1, fleet_server-1.2.0, synthetics-0.10.2
       │ debg Bundling apm-8.4.0-fake-version as apm-8.5.0-SNAPSHOT to align with stack version
       │ debg [1/3] Attempting download of https://epr-snapshot.elastic.co/epr/elastic_agent/elastic_agent-1.3.5.zip 
       │ debg [1/3] Attempting download of https://epr-snapshot.elastic.co/epr/synthetics/synthetics-0.10.2.zip 
       │ debg [1/3] Attempting download of https://epr-snapshot.elastic.co/epr/fleet_server/fleet_server-1.2.0.zip 
       │ debg [1/3] Attempting download of https://epr-snapshot.elastic.co/epr/apm/apm-8.4.0-fake-version.zip 
       │ debg [1/3] Attempting download of https://epr-snapshot.elastic.co/epr/endpoint/endpoint-8.4.1.zip 
       │ debg Downloaded 0 bytes to /Users/kylepollich/Workspace/elastic/kibana/build/kibana/x-pack/plugins/fleet/target/bundled_packages/apm-8.5.0-SNAPSHOT.zip
       │ debg Download failed: Request failed with status code 404
       │ debg Deleting downloaded data at /Users/kylepollich/Workspace/elastic/kibana/build/kibana/x-pack/plugins/fleet/target/bundled_packages/apm-8.5.0-SNAPSHOT.zip
       │ info Retrying in 5 seconds
       │ debg Downloaded 3752 bytes to /Users/kylepollich/Workspace/elastic/kibana/build/kibana/x-pack/plugins/fleet/target/bundled_packages/fleet_server-1.2.0.zip
       │ debg Downloaded https://epr-snapshot.elastic.co/epr/fleet_server/fleet_server-1.2.0.zip 
       │ debg Downloaded 883888 bytes to /Users/kylepollich/Workspace/elastic/kibana/build/kibana/x-pack/plugins/fleet/target/bundled_packages/elastic_agent-1.3.5.zip
       │ debg Downloaded https://epr-snapshot.elastic.co/epr/elastic_agent/elastic_agent-1.3.5.zip 
       │ debg Downloaded 239571 bytes to /Users/kylepollich/Workspace/elastic/kibana/build/kibana/x-pack/plugins/fleet/target/bundled_packages/endpoint-8.4.1.zip
       │ debg Downloaded https://epr-snapshot.elastic.co/epr/endpoint/endpoint-8.4.1.zip 
       │ debg Downloaded 121805 bytes to /Users/kylepollich/Workspace/elastic/kibana/build/kibana/x-pack/plugins/fleet/target/bundled_packages/synthetics-0.10.2.zip
       │ debg Downloaded https://epr-snapshot.elastic.co/epr/synthetics/synthetics-0.10.2.zip 
       │ debg [2/3] Attempting download of https://epr-snapshot.elastic.co/epr/apm/apm-8.4.0-fake-version.zip 
       │ debg Downloaded 0 bytes to /Users/kylepollich/Workspace/elastic/kibana/build/kibana/x-pack/plugins/fleet/target/bundled_packages/apm-8.5.0-SNAPSHOT.zip
       │ debg Download failed: Request failed with status code 404
       │ debg Deleting downloaded data at /Users/kylepollich/Workspace/elastic/kibana/build/kibana/x-pack/plugins/fleet/target/bundled_packages/apm-8.5.0-SNAPSHOT.zip
       │ info Retrying in 10 seconds
       │ debg [3/3] Attempting download of https://epr-snapshot.elastic.co/epr/apm/apm-8.4.0-fake-version.zip 
       │ debg Downloaded 0 bytes to /Users/kylepollich/Workspace/elastic/kibana/build/kibana/x-pack/plugins/fleet/target/bundled_packages/apm-8.5.0-SNAPSHOT.zip
       │ debg Download failed: Request failed with status code 404
       │ debg Deleting downloaded data at /Users/kylepollich/Workspace/elastic/kibana/build/kibana/x-pack/plugins/fleet/target/bundled_packages/apm-8.5.0-SNAPSHOT.zip
       │ERROR Failed to download bundled package archive apm-8.5.0-SNAPSHOT.zip
       │ERROR failure 16 sec
       │ERROR Request failed with status code 404

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
> 404s are indeed concerning. Could you please share more details about where/what have you noticed?

I clicked on random builds until I found this one:
https://buildkite.com/elastic/kibana-artifacts-snapshot/builds/554#0182def8-de84-4f27-8347-8e4e8b0e0d21/198-1593

There's more, and I've had it happen locally too.  Usually just a package that hasn't been uploaded yet.

Reply:
I see, acknowledged errors. We will change the backend of the EPR soon (v1 -> v2), so I guess those will disappear.